src/lsptoolshost/razor/razorEndpoints.ts

```diff
@@ -252,6 +252,13 @@ export function registerRazorEndpoints(
                 return undefined;
             }
 
+            // We know that we've got the right document, and we've been told about the right content by the server,
+            // but all we can be sure of at this point is that we've fired the change event for it. The event firing
+            // is async, and the didChange notification that it would generate is a notification, so doesn't necessarily
+            // block. Before we actually make a call to the Html server, we should at least make sure that the document
+            // version is not still the same as before we updated the content.
+            await document.waitForBufferUpdate();
```

Comment:
I don't know if this applies here.  But it is possible ensure something runs after the didChange (or other lsp notification) fires.  We do that for onAutoInsert because its important that the server receives the didChange before the onautoinsert request.

See https://github.com/dotnet/vscode-csharp/blob/main/src/lsptoolshost/autoInsert/onAutoInsert.ts#L23

Even if you don't block, if the html server is processing LSP requests correctly as long as it receives the didChange first it should process it before everything else in the queue.

Reply:
Oh wait, that would need us to get a reference to the Html language client. I think I've tried that before, and asked VS Code folks, and it's not possible. Will double check tomorrow though.